### PR TITLE
ONNX export: Slice op - Handle None value for ends

### PR DESCRIPTION
**Pull request by vandanavk**
## Description ##
Fixes https://github.com/apache/incubator-mxnet/issues/14875

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Handle ends=None in Slip_axis export to ONNX

## Comments ##
- All ONNX export tests passed

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [ONNX, pr-awaiting-review]

**Comment by Roshrini** [comment]
@vandanavk Thanks for fixing this. Can you please add unit test or model test?

**Comment by karan6181** [comment]
@vandanavk Did you get the chance to add the unit test or model test ? Thanks!

**Comment by piyushghai** [comment]
@vandanavk Gentle ping....

**Comment by vandanavk** [comment]
Working on the test. Will have an update soon.

**Comment by karan6181** [comment]
@vandanavk any updates on the progress?

**Comment by ntomer** [comment]
I am awaiting the changes :)

**Comment by vandanavk** [comment]
This PR is ready for review. [Awaiting confirmation from the user too](https://github.com/apache/incubator-mxnet/issues/14875#issuecomment-513933483)

**Comment by ntomer** [comment]
Any word on when this might pass review?

**Comment by vandanavk** [comment]
@Roshrini @anirudhacharya @zhreshold for review

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 1]
```diff
@@ -1499,17 +1500,19 @@ def convert_slice_axis(node, **kwargs):
 
     axes = int(attrs.get("axis"))
     starts = int(attrs.get("begin"))
-    ends = int(attrs.get("end", None))
-    if not ends:
-        raise ValueError("Slice: ONNX doesnt't support 'None' in 'end' attribute")
+    ends = attrs.get("end", None)
+    if not ends or ends == 'None':
+        # ONNX doesn't support None for ends. Since ends=None depicts
```
Comment (anirudhacharya) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: why should None be mapped to INT_MAX?

Comment (vandanavk) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: I think it is sufficient to map None to the dimension of the particular axis. Changing this in the next iteration.

**Comment by vandanavk** [comment]
Based on offline discussion with @Roshrini, making use of ONNX's "slice_end_out_of_bounds" test instead of writing a custom test

**Comment by vandanavk** [comment]
All review comments addressed. Awaiting further review comments or approval.

**Comment by vandanavk** [comment]
@zhreshold for review/merge

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 1 above)
Comment (vandanavk) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: @anirudhacharya fixed this. the latest commit does not map None to INT_MAX

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py [hunk 2]
```diff
@@ -499,15 +499,19 @@ def split(attrs, inputs, proto_obj):
...
+    input_shape = input_tensor_data[1]
     new_attrs = translation_utils._fix_attribute_names(attrs,
                                                        {'axes' : 'axis',
                                                         'ends' : 'end',
                                                         'starts' : 'begin'})
     # onnx slice provides slicing on multiple axis. Adding multiple slice_axis operator
     # for multiple axes from mxnet
     begin = new_attrs.get('begin')
-    end = new_attrs.get('end')
+    end = list(new_attrs.get('end'))
```
Comment (anirudhacharya) [inline python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: if `end` does not exist in new_attrs, converting it to a list will throw NoneType error

Comment (vandanavk) [reply python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: Its a required field in ONNX, will always exist https://github.com/onnx/onnx/blob/master/docs/Changelog.md#slice-1

**Comment by vandanavk** [comment]
@mxnet-label-bot update [ONNX, pr-awaiting-merge]

**Comment by chouxianyu** [comment]
**Environment:**
- Windows10
- Python3.7
- CPU

**Descroption:**
I met this problem.
And I tried the solution in [**9dcf71d**](https://github.com/apache/incubator-mxnet/commit/9dcf71d8fe33f77ed316a95fcffaf1f7f883ff70), found a new bug.

>Traceback (most recent call last):
  File "C:\Program Files\PyCharm2018.1.4\helpers\pydev\pydevd.py", line 1664, in <module>
    main()
  File "C:\Program Files\PyCharm2018.1.4\helpers\pydev\pydevd.py", line 1658, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "C:\Program Files\PyCharm2018.1.4\helpers\pydev\pydevd.py", line 1068, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Program Files\PyCharm2018.1.4\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "E:/Study/Programming/Code/FWWBA05/MXNet_Helmet_Detection/ONNXTest.py", line 2, in <module>
    export_model('./symbols/darknet53-symbol.json', './symbols/darknet53-0000.params', [(4, 3, 416, 416)], verbose=True)
  File "D:\WorkingSoftware\Anaconda3\lib\site-packages\mxnet\contrib\onnx\mx2onnx\export_model.py", line 83, in export_model
    verbose=verbose)
  File "D:\WorkingSoftware\Anaconda3\lib\site-packages\mxnet\contrib\onnx\mx2onnx\export_onnx.py", line 253, in create_onnx_graph_proto
    idx=idx
  File "D:\WorkingSoftware\Anaconda3\lib\site-packages\mxnet\contrib\onnx\mx2onnx\export_onnx.py", line 90, in convert_layer
    raise AttributeError("No conversion function registered for op type %s yet." % op)
**AttributeError: No conversion function registered for op type _arange yet.**

**Pull request merged**